docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -62,50 +63,65 @@ Below is a quick example of how to setup the a specific stable version of nuclio
+- Set `*.image.pullPolicy` to `Never` or to `IfNotPresent` to ensure that Kubernetes doesn't try to fetch the images from the web.
+- Set `offline` to `true` to put Nuclio in "offline" mode.
+- Set `dashboard.baseImagePullPolicy` to `Never`.
+- Set `registry.pushPullUrl` to a registry URL that's reachable from your system.
+- To use a Nuclio templates library (optional), set `dashboard.templatesArchiveAddress` to a location for serving the templates.
+  Note that you need to package the templates yourself and ensure that they're served locally in a way that's accessible to your system.
 
-- Most definitely [version freeze](#version-freezing). Also, set `*.image.pullPolicy` to `Never` or `IfNotPresent` to make sure k8s won't try to access the web to fetch images at any point in time.
-- Set `Values.offline=true` in the helm values, to put nuclio in "offline" mode. Set `dashboard.baseImagePullPolicy=Never`.
-- Needless to say, in this scenario, you will have to configure nuclio with `registry.pushPullUrl` which is reachable from your system.
-- The processor and onbuild images will also have to be accessible to the dashboard in your environment, as they are required for the building process - (by `docker build`, or [kaniko](#using-kaniko-as-an-image-builder)).
-  This can be tricky as you have to either make those images available to the k8s docker daemon or pull-able from a reachable registry, where they should be preloaded. Use `Values.registy.defaultBaseRegistryURL` to point nuclio at searching in your registry for those images, rather then at the default location of `quay.io/nuclio`.
-  To save some work on setting up a registry and preloading the onbuild images to it (or as a reference to what it should include) - take a look at the [prebaked-registry](https://github.com/nuclio/prebaked-registry).
-- For the Nuclio templates library to be available to you, you'll have to package that yourself, and have it served locally, somewhere within reach of your system. To point Nuclio to it, set `Values.dashboard.templatesArchiveAddress` to where you serve the templates.
+<a id="air-gapped-envir-processor-n-on-build-images"></a>In addition, the processor and on-build images must also be accessible to the dashboard in your environment, as they're required for the build process (either by `docker build` or [Kaniko](#using-kaniko-as-an-image-builder)).
+You can achieve this using either of the following methods:
 
+- Make the images available to the Kubernetes Docker daemon.
+- Preload the images to a registry that's accessible to your system, to allow pulling the images from the registry.
+  When using this method, set `registy.defaultBaseRegistryURL` to the base URL to be used for searching your registry for the images (instead of the default location of `quay.io/nuclio`, which isn't accessible in air-gapped environments).
```

Comment:
```suggestion
  When using this method, set `registy.defaultBaseRegistryURL` to the URL of the registry used for the base and onbuild images (instead of the default location of `quay.io/nuclio`, which isn't accessible in air-gapped environments).
```

Reply:
I thought "base" in `registy.defaultBaseRegistryURL` refers to the URL, not the images, which seemed to align with the "searching ..." text, which I took from the original doc version. I don't think we should say "base and onbuild images" because we don't refer to "base images" anywhere else, we refer to "processor and on-build images". =>
I've now changed it to the following — see commit bdbfe19. I used "the URL of an accessible local registry that contains the preloaded images" (similar to to the related Kaniko doc, which I've now edited to remove "base" — per your separate request — but without the image types, which are already listed in the preceding text); I also edited the text in parentheses for more accuracy:
> When using this method, set `registy.defaultBaseRegistryURL` to the URL of an accessible local registry that contains the preloaded images (thus overriding the default location of `quay.io/nuclio`, which isn't accessible in air-gapped environments)

Reply:
> "we don't refer to "base images" anywhere else, we refer to "processor and on-build images" - base images are a crucial part of building a nuclio function container image and we'll need to refer to them in this doc.
That being said, I've unearthed what looks like a bug in how we refer to those today via this specific value `defaultBaseRegistryURL`, which causes this to only effect onbuild images and not base images.
base images can be controlled via a different "knob" I'll be adding shortly. Naming is not ideal here...
So, let's go with: `to the URL of the registry used for the onbuild images` for now, and i'll add a dedicated bulletin for base images once i'm done fixing / testing

Reply:
Currently, the doc refers to 'the URL of an accessible local registry that contains the preloaded processor and "onbuild" images'. As I understand from you, we shouldn't add the base-images reference at this stage, as it requires a new configuration. I wasn't sure whether you want to keep the reference to the processor images or refer only to the "onbuild" images at this stage, so I kept it for now (as it was used in the original text).
(BTW, I checked and our current terminology in the function configuration reference is '"onbuild" image, "base container image", and "function processor image".)

Reply:
Keep the reference to the processor images 👍

Reply:
Already done ... .